backend-rust/src/graphql_api/module_reference_event.rs

```diff
@@ -139,7 +139,19 @@ impl ModuleReferenceEvent {
             }),
         )?;
 
-        let mut items = sqlx::query_as!(
+        let total_count: u64 = sqlx::query_scalar!(
+            "SELECT
+                COUNT(*)
```

Comment:
Here we can rely on the gapless index as well
```suggestion
                MAX(index) + 1
```